src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -346,6 +333,59 @@ export function transform(formConfig, form) {
     ),
   );
 
+  const claimedConditions = clonedData.ratedDisabilities
+    ? clonedData.ratedDisabilities.map(d => d.name.toLowerCase())
+    : [];
+
+  if (clonedData.newDisabilities) {
+    // Add new disabilities to claimed conditions list
+    clonedData.newDisabilities.forEach(disability =>
+      claimedConditions.push(disability.condition.toLowerCase()),
+    );
+
+    if (clonedData.powDisabilities) {
+      // Add POW specialIssue to new conditions
+      const powDisabilities = transformRelatedDisabilities(
+        clonedData.powDisabilities,
+        claimedConditions,
+      ).map(name => name.toLowerCase());
+      clonedData.newDisabilities = clonedData.newDisabilities.map(d => {
+        if (powDisabilities.includes(d.condition.toLowerCase())) {
+          const newSpecialIssues = (d.specialIssues || []).slice();
+          newSpecialIssues.push(specialIssueTypes.POW);
+          return _.set('specialIssues', newSpecialIssues, d);
+        }
+        return d;
+      });
+
+      delete clonedData.powDisabilities;
+    }
+
+    // Add 'cause' of 'NEW' to new ptsd disabilities since form does not ask
+    const withPtsdCause = clonedData.newDisabilities.map(
+      disability =>
+        disability.condition.toLowerCase().includes('ptsd')
+          ? _.set('cause', causeTypes.NEW, disability)
+          : disability,
+    );
+
+    // Split newDisabilities into primary and secondary arrays for backend
+    const newPrimaryDisabilities = withPtsdCause.filter(
+      disability => disability.cause !== causeTypes.SECONDARY,
+    );
+    const newSecondaryDisabilities = withPtsdCause.filter(
+      disability => disability.cause === causeTypes.SECONDARY,
+    );
+
+    if (newPrimaryDisabilities.length) {
```

Comment:
Will `newPrimaryDisabilities` ever be null/undefined or otherwise not have a length?  Otherwise since `0` is falsey this is a clean way to check 👍

Reply:
If a veteran was only claiming an increase on an existing disability, this would be empty, yeah.

Reply:
In this context, no. This whole chunk only gets run `if (clonedData.newDisabilities)`, where:
1. `const withPtsdCause = clonedData.newDisabilities.map`...
2. `const newPrimaryDisabilities = withPtsdCause.filter(`...

So it could be an empty array, but not undefined/null. This is definitely not awesome because this will break if we rearrange the transformer, but I think refactoring the transformer itself is something that should be done in a separate ticket.